src/axolotl/core/builders/base.py

```
@@ -381,6 +381,18 @@ def _configure_custom_optimizer(
                 optimizer_cls = FlashLion
                 if "betas" in adam_kwargs:
                     optimizer_kwargs["betas"] = adam_kwargs["betas"]
+            elif self.cfg.optimizer == "scao":
+                try:
+                    from scao.optimizer import SCAO
+                except ImportError:
+                    raise ImportError(
+                        "SCAO optimizer not found. Please install it with 'pip install scao'"
+                    )
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)